NIPTool/crud/insert.py

```diff
@@ -48,10 +48,13 @@ def insert_samples(
 def insert_user(adapter: NiptAdapter, user: User) -> str:
     """Function to load a new user to the database."""
 
-    # Check that the username is unique, this should be controlled by creating a index with restrictions
-    existing_user: Optional[dict] = adapter.user_collection.find_one({"username": user.username})
-    if existing_user:
-        raise InsertError(f"User with username {user.username} already exist in database.")
+    existing_user: Optional[dict] = adapter.user_collection.find(
+        {"$or": [{"username": user.username}, {"email": user.email}]}
```

Comment:
is there a way to just set mongo fields to unique? Then you dont need to check, and can just handle db errors